### DISCUSSION TITLE
3172-T current consumption

### DISCUSSION THREAD

**Discussion by srlevitt** [danak6jq/RAK3172#18] [General]
Hi there. Has this code ever been used for a 3172 with TCXO? I'm using such a beast, and the code I have is very similar, apart from the radio target bits'n'bobs. I have my code working, sending, then entering the stop mode, but the current consumption remains stubbornly at around 370uA. I was hoping to get nearer to 20 or so. I swear I'm doing all the same kinds of de-inits as in this code.

**Comment by DominusFulguris** [comment]
You can get the 3172-T much lower than 20uA.

When setup right, the TCXO is only used by the radio so doesn't factor into the stop current - i.e. whatever is causing the higher stop current would also happen with the non TCXO version.

________________________________
From: Steve Levitt ***@***.***>
Sent: Monday, August 19, 2024 8:11:41 PM
To: danak6jq/RAK3172 ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [danak6jq/RAK3172] 3172-T current consumption (Discussion #18)


Hi there. Has this code ever been used for a 3172 with TCXO? I'm using such a beast, and the code I have is very similar, apart from the radio target bits'n'bobs. I have my code working, sending, then entering the stop mode, but the current consumption remains stubbornly at around 370uA. I was hoping to get nearer to 20 or so. I swear I'm doing all the same kinds of de-inits as in this code.

**Comment by danak6jq** [comment]
I do not have access to an RAK3172-T and can not speak specifically for it. However, the RAK3172-SiP contains a TCXO, which, as @DominusFulguris mentions, is switched on/off for the radio via PB0, which is likely the same configuration as the 3172-T.

@srlevitt I know it's been over a year and you've likely solved this; do you have a debugger (like ST/Link v2) attached to the MCU? Even after disabling debug in the MCU configuration (which drops idle Idd from ~2.3mA to ~350uA), you will want to remove the Vdd connection to the debugger. After doing that, I'm seeing vanishingly low idle Idd on my test RAK3172-SiP. I left the other ST/Link connections in place and the LPUART1 console, and I'm seeing < 1uA Idd between wake periods. Measured with a Nordic PPK.
